.cicd/scripts/srw_test.sh

```diff
@@ -17,19 +17,72 @@ else
     workspace="$(cd -- "${script_dir}/../.." && pwd)"
 fi
 
+# Normalize Parallel Works cluster platform value.
+declare platform
+if [[ "${SRW_PLATFORM}" =~ ^(az|g|p)cluster_noaa ]]; then
+    platform='noaacloud'
+else
+    platform="${SRW_PLATFORM}"
+fi
+
 we2e_experiment_base_dir="${workspace}/experiments"
 we2e_test_dir="${workspace}/regional_workflow/tests/WE2E"
 
 we2e_test_file="${we2e_test_dir}/experiments.txt"
 
-# The default set of end-to-end tests to run.
-# TODO: Create a list of additional tests that can be run when a parameter
-# is set to true.
-declare -a we2e_default_tests
-we2e_default_tests=('grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16'
+# The fundamental set of end-to-end tests to run.
+declare -a we2e_fundamental_tests
+we2e_fundamental_tests=('grid_RRFS_CONUS_25km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16'
     'grid_RRFS_CONUS_13km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16'
     'grid_SUBCONUS_Ind_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16')
 
+# The comprehensive set of end-to-end tests to run.
+declare -a we2e_comprehensive_tests
+we2e_comprehensive_tests=('grid_RRFS_CONUS_3km_ics_FV3GFS_lbcs_FV3GFS_suite_GFS_v16'
+    'grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta'
+    'grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta'
+    'grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_RRFS_v1beta'
+    'grid_RRFS_CONUScompact_25km_ics_HRRR_lbcs_RAP_suite_HRRR'
+    'grid_RRFS_CONUScompact_13km_ics_HRRR_lbcs_RAP_suite_HRRR'
+    'grid_RRFS_CONUScompact_3km_ics_HRRR_lbcs_RAP_suite_HRRR'
+)
+
+# The set of workflow tests to run.
+declare -a we2e_workflow_tests
+we2e_workflow_tests=('community_ensemble_008mems'
```

Comment:
Specifying the lists of tests explicitly in this script is ok for the `release/public-v2` branch, but when this is generalized to the `develop` branch, the tests to be run should be read in from some external source (e.g. a file).  That's because a PR itself may change the tests (change test names and add/delete tests).  That more dynamic approach requires adding a test type to each test, e.g. by creating a variable in each test config file named `WE2E_TEST_TYPE` that can be set to `fundamental` or `comprehensive`.  So this script can call another script (e.g. some modified version of `get_WE2Etest_names_subdirs_descs.sh` in `regional_workflow/tests/WE2E`) that outputs a text file identifying the type of each test (fundamental or comprehensive).  This script can then read in that file to determine which tests to run.  

This is just a note for when the automated tests are added to the `develop` branch.  Doesn't affect this PR.

Reply:
I should also add that when the tests are added to the `develop` branch, there needs to be documentation added that describes how to run the tests and a bit of what's happening under the hood (which scripts do what).

Reply:
One more thing, again for when this is generalized to `develop`.  In the PR message, it says "NOTE: The tests results can be found under the S3 Artifacts section of a Jenkins Build."  Does that mean the user needs a Jenkins account that's somehow linked to this repo?  If so, I imagine that can be a bit of a pain to access.  What report comes back to github?  It would be great if at a minimum the list of which tests passed and failed were returned, even better if the the `log.launch_FV3LAM_wflow` log file for each test was returned, and best if the log files from each task were returned so the user can see what exactly went wrong during the test (those log files are under the `${test_name}/log` directory of each test).

Reply:
@gsketefian the users don't need a Jenkins account to view the build log and artifacts. I am capturing the `log.launch_FV3LAM_wflow` in this PR and am working on an update to capture the `log.generate_FV3LAM_wflow` now. I can also include the files under the `${test_name}/log` directory. I may need to put them in all in a tarball for each test to keep things relatively clean. How does that sound?

Reply:
@jessemcfarland That sounds excellent :). I wonder if it would be better to have all the test log files in one tarball instead of each test in a separate one.  That way the user needs to `untar` only one file.  Just a thought, I don't know the answer.  Also, I'm thinking It would be good to have the file that summarizes the final status of each test outside the tarball(s) so the user can just look at it without the need to `untar`.  But we can first try whatever you think is best and make adjustments later as needed.  Thanks for your work on this.

Reply:
@gsketefian you should be able see the artifacts for this latest build here: https://jenkins-epic.woc.noaa.gov/job/ufs-srweather-app/job/pipeline/job/PR-297/9/s3/.

Reply:
@gsketefian 
1. You can reach the pipeline display page by clicking **Details** next to **Build and Test** under the checks section of the PR status below. From that page, you need to switch from the _Blue Ocean_ UI to _Classic_ by clicking the _->_ icon in the top right corner of the page. From the _Classic_ build page, you'll find **S3 Artifacts** on the left navigation menu, assuming there are artifacts for the build.
2. This is configurable, so we can do whatever makes sense. I think this is one topic to discuss in a larger discussion we should have on how the pipeline can be used for the development and release workflows. Right now, the artifacts for the `develop` and release branches are kept indefinitely and artifacts for defunct branches and PRs are kept for 30 days.
3. The intent was to preserve the binaries created by a build so that can be used on the target platforms. My understanding is that the `build` directory is just the "working directory" for the build and wasn't required to use the binaries, is that not the case? You can still examine the `build` directory for the latest build by inspecting the Jenkins _Workspace_.

Reply:
@jessemcfarland:
1. Thanks, I was able to look at the 7 pipeline branches (not to be confused with git branches!) and look at the details under each.  For the documentation, I created an issue in this repo (#[307](https://github.com/ufs-community/ufs-srweather-app/issues/307)) and took the liberty to assgin you and @gspetro to it.  Please let me know (either of you) if that's not ok.
2. I invited you to the Thursday code management meetings (indefinitely for now; can remove you later if you'd like).  No meeting this Thursday, but there should be one next week.
3. Found the build log under Matrix.  Yeah, it would be good to archive it as well so a user can download everything and view it locally instead of having to toggle back and forth between their machine and Jenkins.  Feel free to do it in this PR or another one.
4. Ok.  Don't quite understand why there are three lines named the same (see attachment).  I have to play with this more to understand it better.
<img width="972" alt="Jenkins_workspaces" src="https://user-images.githubusercontent.com/31046882/173874348-c337a810-0bb5-4b84-9690-be575b2e71fb.png">

Reply:
@gsketefian ah ok, that's a good question, I believe there is a workspace link for each stage in the pipeline, init, build, test. For this pipeline, they all happen to point to the same workspace and machine. This would be a more handy feature in another use case where you build on one machine and test on a different machine.